InfiniteLoopBookProject/tests/TestDAO.java

```diff
@@ -1,4 +1,279 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
-public class TestDAO {
+import java.io.IOException;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
 
+import com.st.lms.dao.DAOAuthorImpl;
+import com.st.lms.dao.DAOBookImpl;
+import com.st.lms.dao.DAOPublisherImpl;
+import com.st.lms.model.Author;
+import com.st.lms.model.Book;
+import com.st.lms.model.Publisher;
+
+public class TestDAO {
+	
+	@BeforeClass
+	public static void setUp() throws IOException{
+		AuthorDaoTest.setUp();
+		BookDaoTest.setUp();
+		PublisherDaoTest.setUp();
+	}
+	
+	@AfterClass
+	public static void tearDown() throws IOException{
+		AuthorDaoTest.tearDown();;
+		BookDaoTest.tearDown();;
+		PublisherDaoTest.tearDown();;
+	}
+	
+	public static void testGet() throws IOException{
+		AuthorDaoTest.testGet();
+		BookDaoTest.testGet();
+		PublisherDaoTest.testGet();
+	}
+	
+	public static void testSet() throws IOException{
+		AuthorDaoTest.testSet();
+		BookDaoTest.testSet();
+		PublisherDaoTest.testSet();
+	}
+	
+	public static void testAdd() throws IOException{
+		AuthorDaoTest.testAdd();
+		BookDaoTest.testAdd();
+		PublisherDaoTest.testAdd();
+	}
+	
+	public static void testDelete() throws IOException{
+		AuthorDaoTest.testDelete();
+		BookDaoTest.testDelete();
+		PublisherDaoTest.testDelete();
+	}
+	
+	static class AuthorDaoTest{
+		public static void setUp() throws IOException{
+			DAOAuthorImpl aDao = new DAOAuthorImpl();
+			Author author = new Author();
+			author.setName("JK Rowling");
+			author.setID(2001);
+			aDao.addAuthor(author);
+			author.setName("Masashi Kishimoto");
+			author.setID(2003);
+			aDao.addAuthor(author);
+			author.setName("George RR Martin");
+			author.setID(2005);
+			aDao.addAuthor(author);
+		}
+		
+		public static void tearDown() throws IOException{
+			DAOAuthorImpl aDao = new DAOAuthorImpl();
+			aDao.delete(2001);
+			aDao.delete(2005);
+			aDao.delete(2020);
+		}
+		
+		public static void testGet() throws IOException {
+			DAOAuthorImpl dao = new DAOAuthorImpl();
+			assertTrue(dao.getAuthor(2001) != null);
+			Author author = dao.getAuthor(2001);
+			assertTrue(dao.getAuthor(5000) == null);
+			assertTrue(dao.getAuthor(2001).getName().equals("JK Rowling"));
+			assertTrue(dao.getAuthor(2001).getID() == 2001);

```

Comment:
Use the methods that JUnit provides (e.g. `assertNotNull`, `assertNull`, `assertEquals`) instead of `assertTrue`.